dotCMS/src/main/java/com/dotcms/rest/IntegrityResource.java

```diff
@@ -230,79 +260,138 @@ public Response generateIntegrityData(@Context HttpServletRequest request, @Form
-                                InputStream is = Files.newInputStream(Paths.get(
-                                        ConfigUtils.getIntegrityPath() + File.separator
-                                                + requesterEndPoint.getId() + File.separator
-                                                + INTEGRITY_DATA_TO_CHECK_ZIP_FILE_NAME));
-
-                                byte[] buffer = new byte[1024];
-                                int bytesRead;
-                                //read from is to buffer
-                                while((bytesRead = is.read(buffer)) !=-1){
-                                    output.write(buffer, 0, bytesRead);
-                                }
-                                is.close();
-                                //flush OutputStream to write any buffered data to file
-                                output.flush();
-                                output.close();
-
-                            }
-                        };
-                        return Response.ok(output).build();
-
-                    case CANCELED:
-                        return Response.status( HttpStatus.SC_RESET_CONTENT ).entity( servletContext.getAttribute( "integrityDataGenerationError" ) ).build();
-
-                    case ERROR:
-                        return Response.status(HttpStatus.SC_INTERNAL_SERVER_ERROR).entity(servletContext.getAttribute("integrityDataGenerationError")).build();
+            if (integrityMetadata.get().getProcessStatus() == ProcessStatus.PROCESSING) {
+                Logger.info(
+                        IntegrityResource.class,
+                        String.format(
+                                "Receiver at %s:> integrity data generation for endpoint id %s still ongoing therefore it's not ready and need to wait",
+                                localAddress,
+                                requesterEndpoint.getId()));
+                return Response.status(HttpStatus.SC_PROCESSING).build();
+            } else if (integrityMetadata.get().getProcessStatus() == ProcessStatus.FINISHED &&
+                    IntegrityUtil.doesIntegrityDataFileExist(
+                            requesterEndpoint.getId(),
+                            IntegrityUtil.INTEGRITY_DATA_TO_CHECK_ZIP_FILENAME)) {
+                final String zipFilePath = IntegrityUtil.getIntegrityDataFilePath(
+                        requesterEndpoint.getId(),
+                        IntegrityUtil.INTEGRITY_DATA_TO_CHECK_ZIP_FILENAME);
+                final StreamingOutput output = so -> {
+                    final InputStream inputStream = Files.newInputStream(Paths.get(zipFilePath));
+                    final byte[] buffer = new byte[1024];
+                    int bytesRead;
+                    //read from is to buffer
+                    while((bytesRead = inputStream.read(buffer)) != -1){
```

Comment:
Codacy found an issue: [Avoid assignments in operands](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6151243)